### PR TITLE
automation for testing quota management

### PR DESCRIPTION
**Pull request by hmaheswa**
Automating quota management testing in rgw/v2.
Logs: http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-MFBSVA/

Signed-off-by: Hemanth Maheswarla <hmaheswa@hmaheswa.remote.csb>

### CODE REVIEW

**Review by udaysk23 — COMMENTED** [review]
Please attach the log of successful run

**Comment by hmaheswa** [comment]
here is the log of successful run:
https://159.23.92.24/job/rhceph-test-execution-manual/43/consoleFull

File: rgw/v2/tests/s3_swift/test_quota_management.py [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+    )
+    reusable.enable_bucket_quota(user_info)
+
+    try:
+        upload_object_initiate(
+            config, user_info, bucket, bucket_name, 1, config.bucket_max_size + 100
+        )
+    except TestExecError as e:
+        log.info("bucket quota with max size passed")
+    except Exception:
```
Comment (anrao19) [inline rgw/v2/tests/s3_swift/test_quota_management.py:95]: We can have
except:
        raise AssertionError("bucket quota with max size failed")
  Then we dont have to provide sys.exit(1) and we can skip else part

File: rgw/v2/tests/s3_swift/test_quota_management.py [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+        log.error("bucket quota with max size failed")
+        sys.exit(1)
+
+    for i in range(config.bucket_max_objects):
+        upload_object_initiate(config, user_info, bucket, bucket_name, i, 0)
+    try:
+        upload_object_initiate(config, user_info, bucket, bucket_name, i, 0)
+    except TestExecError as e:
+        log.info("bucket quota with max objects passed")
+    except Exception:
```
Comment (anrao19) [inline rgw/v2/tests/s3_swift/test_quota_management.py:108]: we can have same changes as above comment

File: rgw/v2/tests/s3_swift/test_quota_management.py [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+    reusable.enable_user_quota(user_info)
+
+    try:
+        upload_object_initiate(
+            config, user_info, bucket, bucket_name, 1, config.user_max_size + 100
+        )
+        raise Exception("user quota with max size failed")
+    except TestExecError as e:
+        log.info("user quota with max size passed")
+    except Exception:
```
Comment (anrao19) [inline rgw/v2/tests/s3_swift/test_quota_management.py:128]: Same comment as above

File: rgw/v2/tests/s3_swift/test_quota_management.py [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+        sys.exit(1)
+
+    for i in range(config.user_max_objects):
+        upload_object_initiate(config, user_info, bucket, bucket_name, i, 0)
+    try:
+        upload_object_initiate(config, user_info, bucket, bucket_name, i, 0)
+        raise Exception("user quota with max objects failed")
+    except TestExecError as e:
+        log.info("user quota with max objects passed")
+    except Exception:
```
Comment (anrao19) [inline rgw/v2/tests/s3_swift/test_quota_management.py:142]: Same comment as above

File: rgw/v2/tests/s3_swift/configs/test_quota_management.yaml [hunk 5]
```diff
@@ -0,0 +1,9 @@
+# upload type: non multipart
+# script: test_Mbuckets_with_Nobjects.py
```
Comment (ckulal) [inline rgw/v2/tests/s3_swift/configs/test_quota_management.yaml:2]: please change the script name

Comment (ckulal) [inline rgw/v2/tests/s3_swift/configs/test_quota_management.yaml:2]: please change the script name

**Review by anrao19 — APPROVED** [review]
LGTM

File: rgw/v2/tests/s3_swift/test_quota_management.py [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+        config.read()
+        test_exec(config)
+        test_info.success_status("test passed")
+        sys.exit(0)
+
+        # todo: Verify code to be executed after rename lib changes
+        # Verify data
+        # read_io = ReadIOInfo()
+        # read_io.verify_io()
+
```
Comment (udaysk23) [inline rgw/v2/tests/s3_swift/test_quota_management.py:163]: Please remove this code

**Review by udaysk23 — CHANGES_REQUESTED** [review]

File: rgw/v2/tests/s3_swift/test_quota_management.py [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+        upload_object_initiate(
+            config, user_info, bucket, bucket_name, 1, config.user_max_size + 100
+        )
+        raise AssertionError("user quota with max size failed")
+    except TestExecError as e:
+        log.info("user quota with max size passed")
+
+    for i in range(config.user_max_objects):
+        upload_object_initiate(config, user_info, bucket, bucket_name, i, 0)
+    try:
```
Comment (mkasturi18) [inline rgw/v2/tests/s3_swift/test_quota_management.py:116]: test_exec function is called in try catch block. We can avoid using it again

**Review by udaysk23 — APPROVED** [review]

**Review by udaysk23 — APPROVED** [review]

File: rgw/v2/tests/s3_swift/test_quota_management.py [hunk 8]
```diff
@@ -0,0 +1,210 @@
+"""test_quota_management - Test quota management
+
+Usage: test_quota_management.py -c <input_yaml>
+
+<input_yaml>
+        Note: any one of these yamls can be used
+        test_quota_management.yaml
```
Comment (udaysk23) [inline rgw/v2/tests/s3_swift/test_quota_management.py:7]: Please add all supported yaml list here for future referance

Comment (hmaheswa) [reply rgw/v2/tests/s3_swift/test_quota_management.py]: done uday

**Review by udaysk23 — DISMISSED** [review]

File: rgw/v2/tests/s3_swift/test_quota_management.py [hunk 9]
```diff
@@ -0,0 +1,209 @@
...
+            config,
+            each_user,
+        )
+        return True
+    except TestExecError as e:
+        log.info(e)
+        return False
+
+
+def test_max_objects(quota_scope, config, each_user, bucket, max_objects):
```
Comment (mkasturi18) [inline rgw/v2/tests/s3_swift/test_quota_management.py:67]: @hmaheswa , could we move this also to resuables/quota_management.py

File: rgw/v2/tests/s3_swift/test_quota_management.py [hunk 10]
```diff
@@ -0,0 +1,209 @@
...
+    if not uploaded:
+        raise AssertionError(
+            f"object upload refused even after disabling {quota_scope} quota"
+        )
+    log.info("object upload successful after disabling quota")
+    reusable.delete_objects(bucket)
+    log.info(f"{quota_scope} quota with max objects passed")
+
+
+def test_max_size(quota_scope, config, each_user, bucket, max_size):
```
Comment (mkasturi18) [inline rgw/v2/tests/s3_swift/test_quota_management.py:101]: @hmaheswa , could we move this also to resuables/quota_management.py

Comment (hmaheswa) [reply rgw/v2/tests/s3_swift/test_quota_management.py]: done madhavi

File: rgw/v2/tests/s3_swift/test_quota_management.py (hunk 9 above)
Comment (hmaheswa) [reply rgw/v2/tests/s3_swift/test_quota_management.py]: done madhavi

**Review by mkasturi18 — APPROVED** [review]

**Pull request merged**